File: CodeForce/Contest/CodeforcesRound #692 (Div. 2)/B/awholenewworld.cxx
// Accepted --- 202ms https://codeforces.com/contest/1465/submission/101922403
#include <iostream>
using namespace std;

bool ok(long long a)
{
    long long b = a;
    while (b)
    {
        long long rem = b % 10;
        b /= 10;
        if (rem != 0 && a % rem != 0)
            return false;
    }
    return true;
}

void solve()
{
    long long num;
    cin >> num;
    while (true)
    {
        if (ok(num))
        {
            cout << num << '\n';
            return;
        }
        num++;
    }
}

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);
    int t;
    cin >> t;
    while (t--)
        solve();
}

File: CodeForce/cfr/692_div2/C/awholenewworld.js
// 12.20 evening

// Accepted --- https://codeforces.com/contest/1465/submission/101927984
let MAX = 100005;
let con = 0;
let par = Array(MAX).fill(0);
const parent = (n) => {
    if (par[n] == n) return n;
    return par[n] = parent(par[n]);
};

const join = (a, b) => {
    a = parent(a);
    b = parent(b);
    if (a == b) {
        con++;
        return;
    }
    par[a] = b;
};

const solve = (n, m, a) => {
    con = 0;
    for (let i = 0; i <= n; i++) {
        par[i] = i;
    }
    let req = 0;
    for (let i = 0; i < m; i++) {
        let x = a[i][0];
        let y = a[i][1];
        if (x == y) continue;
        join(x, y);
        req++;
    }
    console.log(req + con);
};

const main = () => {
    const readline = require('readline');
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });
    let input = [];
    rl.on('line', (line) => {
        input.push(line.split(" ").map(x => Number(x)));
    });
    rl.on('close', () => {
        let t = input[0][0];
        let i = 1;
        while (t--) {
            let n = input[i][0];
            let m = input[i][1];
            solve(n, m, input.slice(i + 1, i + m + 1));
            i += m + 1;
        }
    });
};

main()